### PR TITLE
scipy.io.idl: make reader more robust to missing variables in heap

### LINKED ISSUE #4613: scipy.io.readsav error on reading sav file
scipy.io.readsav doesn't work many sav files, such as this one https://dl.dropboxusercontent.com/u/22819/1066671952_cal.sav

Python 2.7.3 (default, Jan 13 2013, 09:19:58)
[GCC 4.4.6 20120305 (Red Hat 4.4.6-4)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import scipy
>>> scipy.__version__
'0.12.0'
>>> import scipy.io
>>> scipy.io.readsav('1066657560_cal.sav')
/usr/local/lib/python2.7/site-packages/scipy/io/idl.py:167: UserWarning: warning: empty strings are now set to '' instead of None
  warnings.warn("warning: empty strings are now set to '' instead of None")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/scipy/io/idl.py", line 813, in readsav
    replace, new = _replace_heap(r['data'], heap)
  File "/usr/local/lib/python2.7/site-packages/scipy/io/idl.py", line 601, in _replace_heap
    replace, new = _replace_heap(record, heap)
  File "/usr/local/lib/python2.7/site-packages/scipy/io/idl.py", line 613, in _replace_heap
    replace, new = _replace_heap(value, heap)
  File "/usr/local/lib/python2.7/site-packages/scipy/io/idl.py", line 627, in _replace_heap
    replace, new = _replace_heap(variable.item(iv), heap)
  File "/usr/local/lib/python2.7/site-packages/scipy/io/idl.py", line 587, in _replace_heap
    variable = heap[variable.index]
KeyError: 543708

### PR DESCRIPTION
**Pull request by astrofrog**
Fixes #4613

### CODE REVIEW

**Comment by astrofrog** [comment]
I am going to see if it's possible to add a regression test for this

File: scipy/io/idl.py [hunk 1]
```diff
@@ -592,7 +593,11 @@ def _replace_heap(variable, heap):
             if variable.index == 0:
                 variable = None
             else:
-                variable = heap[variable.index]
+                if variable.index in heap:
+                    variable = heap[variable.index]
+                else:
+                    warnings.warn("Variable referenced by pointer not found in heap: variable will be set to None")
```
Comment (rgommers) [inline scipy/io/idl.py]: Can you break this line to <80 chars?

Comment (astrofrog) [inline scipy/io/idl.py]: Yes - I'll do that and add a regression test this week.

**Comment by astrofrog** [comment]
(Sorry I got delayed since I am organizing a conference next week - will get back to this soon)

**Comment by rgommers** [comment]
@astrofrog no worries, 6 days is not a long time for a PR to be open in Scipy land...

File: scipy/io/idl.py (hunk 1 above)
Comment (astrofrog) [inline scipy/io/idl.py]: Fixed

**Comment by astrofrog** [comment]
@rgommers - this is ready for a final review

**Comment by rgommers** [comment]
Thanks @astrofrog, merged in 100ce74. 

I made some minor tweaks (replaced ``assert`` with ``assert_``, docstring --> comment in test function, and edited commit message).

**Pull request closed without merging**